include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,257 @@
+	MICRO_STEP_256,
+};
+
+/**
+ * @typedef stepper_motor_controller_set_steps_to_move_t
+ * @brief Stepper Motor Controller shall move the motor with given steps
+ */
+typedef int32_t (*stepper_motor_controller_set_steps_to_move_t)(const struct device *dev,
+								int32_t steps);
+
+/**
+ * @typedef stepper_motor_controller_set_velocity_t
+ * @brief Set the velocity in steps per seconds. For controllers such as DRV8825 where you
+ * toggle the STEP Pin, the pulse_length would have to be calculated based on this parameter in the
+ * driver. For controllers where velocity can be set, this parameter corresponds to max_velocity
+ */
+typedef int32_t (*stepper_motor_controller_set_velocity_t)(const struct device *dev,
+							   uint32_t steps_per_sec);
+
+/**
+ * @typedef stepper_motor_controller_set_micro_step_res_t
+ * @brief Set the microstep resolution
+ */
+typedef int32_t (*stepper_motor_controller_set_micro_step_res_t)(
+	const struct device *dev, enum micro_step_resolution resolution);
+
+/**
+ * @typedef stepper_motor_controller_set_reference_t
+ * @brief reset the actual position in a stepper motor controller
+ */
+typedef int32_t (*stepper_motor_controller_set_reference_t)(const struct device *dev,
+							    int32_t steps);
+
+/**
+ * @typedef stepper_motor_controller_get_position_t
+ * @brief Get the actual position from stepper motor controller
+ */
+typedef int32_t (*stepper_motor_controller_get_position_t)(const struct device *dev,
+							   int32_t *position);
+
+/**
+ * @typedef stepper_motor_controller_set_position_t
+ * @brief Set the target position in stepper motor controller which should set stepper motor in
+ * motion
+ */
+typedef int32_t (*stepper_motor_controller_set_position_t)(const struct device *dev, int32_t steps);
+
+/**
+ * @typedef stepper_motor_controller_enable_velocity_mode_t
+ * @brief Stepper motor controller shall activate the mode where the motor reaches a constant
+ * velocity and runs incessantly
+ */
+typedef int32_t (*stepper_motor_controller_enable_velocity_mode_t)(const struct device *dev,
+								   int32_t constant_velocity);
+
+/**
+ * @typedef stepper_motor_controller_get_stall_status_t
+ * @brief Get the status if a motor is stalled or not from stepper motor controller
+ */
+typedef int32_t (*stepper_motor_controller_get_stall_status_t)(const struct device *dev,
+							       bool *stall_status);
+
+/**
+ * @typedef stepper_motor_controller_enable_stall_detection_t
+ * @brief Enable or Disable stall detection
+ */
+typedef int32_t (*stepper_motor_controller_enable_stall_detection_t)(const struct device *dev,
+								     bool enable_stall_detection);
+
+/**
+ * @typedef stall_detection_motor_controller_set_stall_detection_threshold_t
+ * @brief Set the stall detection threshold
+ */
+typedef int32_t (*stall_detection_motor_controller_set_stall_detection_threshold_t)(
+	const struct device *dev, int32_t stall_detection_threshold);
+
+__subsystem struct stepper_motor_controller_api {
```

Comment:
The api is lacking a simple stepper_motor_controller_enable(.. bool enable ) function to me.
Would this be an applicable change?